src/include/rbtree.h

```diff
@@ -1,17 +1,165 @@
+/*
+ * The main functionality of the red-black trees is to provide a fast, 
+ * cheap way to map ranges of addresses to a number and acccessing by providing an address in this range.
+ */
```

Comment:
Use more detailed wording for the rationale of utilizing red-black trees for memory management. You can discuss the prior situation when red-black trees were not introduced, where sparse arrays were used, resulting in memory waste. With the introduction of red-black trees, explain the considerations and benefits below.

Make it informative.

Reply:
Memory was one goal for sure. But speed of access was also another one. The old method using sparse arrays was an `O(n)` complexity for accessing data, while the rbtree is `O(ln(n)` and because the `memprot` can have hundreds of entries and is used quite often, the speed difference is measurable there.